### DISCUSSION TITLE
Possible to add Png metadata in Chunks? Baking images?

### DISCUSSION THREAD

**Discussion by siva538** [SixLabors/ImageSharp#1525] [Q&A] [answered]
Hello ImageSharp team,

Looking for a feasibility of adding chunks to the Png file as form of metadata. To be specific looking to add iTXt chunk for open badges. Something similar to the baking process - https://github.com/concentricsky/openbadges-bakery

Thanks in advance!

**Comment by brianpopow** [comment, marked as answer]
you can add text data to png images like this:

```
using (var image = Image.Load(@"foobar.png"))
{
    PngMetadata inputMetadata = image.Metadata.GetFormatMetadata(PngFormat.Instance);
    inputMetadata.TextData.Add(new PngTextData("keyword", "example", "en-uk", "text"));
    image.Save("output.png");
}
```

I do not understand what you mean by "baking images"

**Reply by siva538** [reply]
Thanks @brianpopow . This worked like a charm. We would have to convert jpg to png etc,. but am hoping it would be straightforward.  Thanks a lot for your help.

**Reply by JimBobSquarePants** [reply]
[`Image.SaveAsPng(...)`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.ImageExtensions.html#SixLabors_ImageSharp_ImageExtensions_SaveAsPng_SixLabors_ImageSharp_Image_System_IO_Stream_) is an explicit way of doing this.
